src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -86,4 +110,122 @@ public function update(OrderReturn $orderReturn): void
+                'Can\'t delete last product from merchandise return',
+                DeleteOrderReturnProductException::LAST_ORDER_RETURN_PRODUCT
+            );
+        }
+        $orderReturnDetail = $this->getOrderReturnDetailByOrderDetailId($orderReturnDetailId->getValue());
+
+        $this->deleteOrderReturnDetail(
+            $orderReturnDetail->getOrderReturnId(),
+            $orderReturnDetailId,
+            $orderReturnDetail->getCustomizationId()
+        );
+    }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     * @param CustomizationId|null $customizationId
+     *
+     * @throws DeleteOrderReturnProductException
+     */
+    public function deleteOrderReturnDetail(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId,
+        ?CustomizationId $customizationId = null
+    ): void {
+        try {
+            if (!OrderReturn::deleteOrderReturnDetail(
+                $orderReturnId->getValue(),
+                $orderReturnDetailId->getValue(),
+                $customizationId ? $customizationId->getValue() : 0
+            )) {
+                throw new DeleteOrderReturnProductException(
+                    'Failed to delete merchandise return detail',
+                    DeleteOrderReturnProductException::UNEXPECTED_ERROR
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new DeleteOrderReturnProductException(
+                'Failed to delete merchandise return detail',
+                DeleteOrderReturnProductException::UNEXPECTED_ERROR
+            );
+        }
+    }
+
+    /**
+     * @param int $orderDetailId
+     *
+     * @return OrderReturnDetail
+     *
+     * @throws OrderReturnDetailNotFoundException
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function getOrderReturnDetailByOrderDetailId(int $orderDetailId): OrderReturnDetail
+    {
+        $result = $this->connection->createQueryBuilder()
+            ->select('id_order_return, id_order_detail, id_customization, product_quantity')
+            ->from($this->dbPrefix . 'order_return_detail')
+            ->where('id_order_detail = :orderDetailId')
+            ->setParameter('orderDetailId', $orderDetailId)
+            ->execute()->fetchAssociative();
+        if (!$result) {
+            throw new OrderReturnDetailNotFoundException(sprintf('Order return detail with id "%s" not found', $orderDetailId));
+        }
+
+        return new OrderReturnDetail(
+            (int) $result['id_order_return'],
+            (int) $result['id_order_detail'],
+            (int) $result['product_quantity'],
+            (int) $result['id_customization'] ?: null
+        );
+    }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param Order $order
+     *
+     * @return array<OrderReturnDetail>
```

Comment:
Im not sure current definition is legit, I cannot find it in psalm docs. (why I doubt is because usually the first item of the annotation is the key and the second one is value, so it should look like array<int, OrderReturnDetail>). Anyway if you know this way is legit, please just provide me some link/argument so I know for future too :smile: 
```suggestion
     * @return OrderReturnDetail[]
```

Reply:
Agreed, although PHPStan doesn't seem to complain so maybe it's fine